src/config/backends/splash-image.ts

```diff
@@ -0,0 +1,196 @@
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import { fs } from 'mz';
+import * as path from 'path';
+
+import * as constants from '../../lib/constants';
+import log from '../../lib/supervisor-console';
+import {
+	bootMountPoint,
+	ConfigBackend,
+	ConfigOptions,
+	remountAndWriteAtomic,
+} from './backend';
+
+export class SplashImage extends ConfigBackend {
+	private static readonly BASEPATH = path.join(bootMountPoint, 'splash');
+	private static readonly DEFAULT = path.join(
+		SplashImage.BASEPATH,
+		'balena-logo-default.png',
+	);
+	private static readonly FILENAMES = ['balena-logo.png', 'resin-logo.png'];
+	private static readonly PREFIX = `${constants.hostConfigVarPrefix}SPLASH_`;
+	private static readonly CONFIGS = ['image'];
+	private static readonly DATA_URI_REGEX = /^data:(.+);base64,(.*)$/;
+
+	// Check the first 8 bytes of a buffer for a PNG header
+	// Source: https://github.com/sindresorhus/is-png/blob/master/index.js
+	private isPng(buffer: Buffer) {
+		if (!buffer || buffer.length < 8) {
+			return false;
+		}
+
+		return (
+			buffer[0] === 0x89 &&
+			buffer[1] === 0x50 &&
+			buffer[2] === 0x4e &&
+			buffer[3] === 0x47 &&
+			buffer[4] === 0x0d &&
+			buffer[5] === 0x0a &&
+			buffer[6] === 0x1a &&
+			buffer[7] === 0x0a
+		);
+	}
+
+	// Get the file path for the splash image for the underlying
+	// system
+	private async getSplashPath(): Promise<string> {
+		const [
+			// If no logo is defined, assume the file is `balena-logo.png`
+			splashFile = path.join(SplashImage.BASEPATH, 'balena-logo.png'),
+			// Read the splash dir (will throw if the path does not exist)
+		] = await Bluebird.resolve(fs.readdir(SplashImage.BASEPATH))
+			// Check for files in the valid splash filename list
+			.filter((filename) => SplashImage.FILENAMES.includes(filename))
+			// Convert to full path
+			.map((filename) => path.join(SplashImage.BASEPATH, filename));
+
+		return splashFile;
+	}
+
+	// Returns the base64 contents of the splash image
+	private async readSplashImage(where?: string): Promise<string> {
+		// Read from defaultPath unless where is defined
+		where = where ?? (await this.getSplashPath());
+
+		// read the image file...
+		return (await fs.readFile(where)).toString('base64');
+	}
+
+	// Write a splash image provided as a base64 string
+	private async writeSplashImage(image: string, where?: string): Promise<void> {
+		// Read from defaultPath unless where is defined
+		where = where ?? (await this.getSplashPath());
```

Comment:
If resin-logo.png exists then images will continue to write to that location. We could take this opportunity to move on from that name and only write to balena-logo.png.

Reply:
It depends on the OS, newer versions of the OS support `balena-logo`, but we might want to have the supervisor to still work with older versions

Reply:
Also noticed the comment says read from defaultPath which it should say Write to defaultPath :p

Reply:
Adding the version check you mentioned would be useful then so we have something in place that is actively moving away from resin naming conventions.

Reply:
"A better way would probably be to check os version and always use balena-logo if >= 2.52 and resin-logo if < 2.52"

Reply:
no longer needed since we are sorting the files to always use balena-logo if it exists thus achieving the affect I wanted here.